examples/tensorflow-akka/src/main/scala/modelserving/TensorFlowModel.scala

```diff
@@ -0,0 +1,171 @@
+package modelserving
+
+import java.io.File
+import java.nio.file._
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.{ Map ⇒ MMap }
+import scala.util.control.NonFatal
+import scala.util.Try
+
+import com.google.protobuf.Descriptors
+import org.tensorflow.framework._
+import org.tensorflow.{ Graph, SavedModelBundle, Session, Tensor }
+import modelserving.wine.avro._
+
+/**
+ * Loads a model from a TensorFlow SavedModelBundle
+ * @param modelName the name of the model
+ * @param savedModelBundlePath the path to the SavedModelBundle
+ * @param createModel a function to create a model from a LoadedModel
+ */
+object TensorFlowModelBundle {
+  def load[Record, ServingResult, Model](
+      savedModelBundlePath: Path,
+      modelName: String,
+      createModel: LoadedModel ⇒ Model
+  ): Try[Model] = {
+    Try {
+      // TODO fix, don't know what these tags are for.
+      // get tags. We assume here that the first tag is the one we use
+      val tags: Seq[String] = getTags(savedModelBundlePath)
+      val bundle = SavedModelBundle.load(savedModelBundlePath.toAbsolutePath.toString, tags(0))
+      val graph = bundle.graph
+      // get metatagraph and signature
+      val metaGraphDef = MetaGraphDef.parseFrom(bundle.metaGraphDef)
+      val signatureMap = metaGraphDef.getSignatureDefMap.asScala
+      //  parse signature, so that we can use definitions (if necessary) programmatically in score method
+      // Create TensorFlow session
+      val session = bundle.session
+      createModel(LoadedModel(modelName, graph, session, parseSignatures(signatureMap)))
+    }
+  }
+
+  /**
+   * Parse signatures
+   *
+   * @param signatures - signatures from metagraph
+   * @returns map of names/signatures
+   */
+  private def parseSignatures(signatures: MMap[String, SignatureDef]): Map[String, Signature] = {
+    signatures.map(signature ⇒
+      signature._1 -> Signature(parseInputOutput(signature._2.getInputsMap.asScala), parseInputOutput(signature._2.getOutputsMap.asScala))).toMap
+  }
+
+  /**
+   * TODO ask about these tags
+   * Gets all tags in the saved bundle and uses the first one. If you need a specific tag, overwrite this method
+   * With a seq (of one) tags returning desired tag.
+   *
+   * @param directory - directory for saved model
+   * @returns sequence of tags
+   */
+  private def getTags(directory: Path): Seq[String] = {
+
+    val d = directory.toFile
```

Comment:
nit: maybe use a more meaningful variable name.

Reply:
Thanks, yeah this is taken over verbatim, I'll improve this more

Reply:
fixed.